ehitchhiking-ui/src/app/shared/modals/profile-modal/profile-modal.component.ts

```diff
@@ -42,22 +43,45 @@ export class ProfileModalComponent implements OnInit {
 		this.addCarMod = !this.addCarMod;
 	}
 
-	onSubmitNewCar(newCar: Car): void {
-		this.user.addCar(newCar);
-		this.carsInfoForm = this.carsInfoService.toFormGroup(this.user.cars, this.formBuilder);
-		this.addCarMod = false;
+	onSubmitNewCar(newCar: any): void {
+		this.isLoading = true;
+		this.profileModalService.addNewCar(newCar, this.currentUser.id).then((car) => {
+			if (car) {
+				this.currentUser.addCar(car);
+				this.carsInfoForm = this.carsInfoService.toFormGroup(this.currentUser.cars, this.formBuilder);
+			}
+			this.addCarMod = false;
+			this.isLoading = false;
+		});
```

Comment:
pls use finally block to disable loading

`then((car) => {...}).finally(() => {this.isLoading = false;})`